test/unit/error.test.ts

```diff
@@ -424,4 +397,147 @@ describe('MongoErrors', () => {
+          error: new MongoError('do not retry me!'),
+          maxWireVersion: 2.3
+        },
+        {
+          description: 'a MongoWriteConcernError with no code nor label',
+          result: false,
+          error: new MongoWriteConcernError({ message: 'empty wc error' }),
+          maxWireVersion: 2.3
+        },
+        {
+          description: 'a MongoWriteConcernError with a random label',
+          result: false,
+          error: new MongoWriteConcernError(
+            { message: 'random label' },
+            { errorLabels: ['myLabel'] }
+          ),
+          maxWireVersion: 0
+        },
+        {
+          description: 'a MongoWriteConcernError with an ExceededTimeLimit code above server 4.4',
+          result: false,
+          error: new MongoWriteConcernError({}, { code: 262 }),
+          maxWireVersion: 9
+        },
+        {
+          description: 'a MongoWriteConcernError with an ExceededTimeLimit code below server 4.4',
+          result: true,
+          error: new MongoWriteConcernError({}, { code: 262 }),
+          maxWireVersion: 2.3
+        },
+        {
+          description: 'a MongoWriteConcernError with a RetryableWriteError label below server 4.4',
+          result: true,
+          error: new MongoWriteConcernError({}, { errorLabels: ['RetryableWriteError'] }),
+          maxWireVersion: 0
+        },
+        {
+          description: 'a MongoWriteConcernError with a RetryableWriteError label above server 4.4',
+          result: true,
+          error: new MongoWriteConcernError({}, { errorLabels: ['RetryableWriteError'] }),
+          maxWireVersion: 9
+        },
+        {
+          description: 'any MongoError with a RetryableWriteError label',
+          result: true,
+          error: (() => {
+            // These tests all use MongoWriteConcernError because
+            // its constructor is easier to call but any MongoError should work
+            const error = new MongoError('');
+            error.addErrorLabel('RetryableWriteError');
+            return error;
+          })(),
+          maxWireVersion: 9
+        }
+      ];
+      for (const { description, result, error, maxWireVersion } of tests) {
+        it(`${description} is${result ? '' : ' not'} a retryable write`, () => {
+          expect(isRetryableWriteError(error, maxWireVersion)).to.be.equal(result);
+        });
+      }
+    });
+
+    describe('#isRetryableReadError', () => {
+      const tests: { description: string; result: boolean; error: MongoError }[] = [
+        {
+          description: 'plain error',
+          result: false,
+          // @ts-expect-error: passing in a plain error to test false case
+          error: new Error('do not retry me!')
+        },
+        {
+          description: 'error code 1',
+          result: false,
+          error: new MongoServerError({ message: '', code: 1 })
+        },
+        {
+          description: 'error code ShutdownInProgress',
```

Comment:
what's special about ShutdownInProgress? there is no special block to handle this error in the code, so it's strange to call it out

Reply:
fixed.